### DISCUSSION TITLE
Book 1 Chapter 5.2: `hit_sphere` function

### DISCUSSION THREAD

**Discussion by LollipopFt** [RayTracing/raytracing.github.io#1057] [Q&A] [answered]
in the last line of `hit_sphere`, the code is:
```c
return (discriminant > 0);
```
however, earlier in the same chapter the book mentioned
> either positive (meaning two real solutions), negative (meaning no real solutions), or zero (meaning one real solution)

doesn't this mean the code should instead be
```c
return (discriminant >= 0)
```
since `discriminant = 0` means that the ray still touches the sphere and should thus be included?

**Comment by hollasch** [comment, marked as answer]
Good catch! Promoted to issue, will get it fixed in v4.0.0.

**Comment by hollasch** [comment]
See issue #1058
